aws/resource_aws_sagemaker_endpoint_configuration.go

```diff
@@ -0,0 +1,278 @@
+	_, err = conn.CreateEndpointConfig(createOpts)
+	if err != nil {
+		return fmt.Errorf("error creating Sagemaker endpoint configuration: %s", err)
+	}
+	d.SetId(name)
+
+	return resourceAwsSagemakerEndpointConfigurationRead(d, meta)
+}
+
+func resourceAwsSagemakerEndpointConfigurationRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	request := &sagemaker.DescribeEndpointConfigInput{
+		EndpointConfigName: aws.String(d.Id()),
+	}
+
+	endpointConfig, err := conn.DescribeEndpointConfig(request)
+	if err != nil {
+		if sagemakerErr, ok := err.(awserr.Error); ok && sagemakerErr.Code() == "ValidationException" {
+			log.Printf("[INFO] unable to find the sagemaker endpoint configuration resource and therefore it is removed from the state: %s", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("error reading Sagemaker endpoint configuration %s: %s", d.Id(), err)
+	}
+
+	if err := d.Set("arn", endpointConfig.EndpointConfigArn); err != nil {
+		return err
+	}
+	if err := d.Set("name", endpointConfig.EndpointConfigName); err != nil {
+		return err
+	}
+	if err := d.Set("production_variants", flattenProductionVariants(endpointConfig.ProductionVariants)); err != nil {
+		return err
+	}
+	if err := d.Set("kms_key_id", endpointConfig.KmsKeyId); err != nil {
+		return err
+	}
+
+	tagsOutput, err := conn.ListTags(&sagemaker.ListTagsInput{
+		ResourceArn: endpointConfig.EndpointConfigArn,
+	})
+	if err != nil {
+		return fmt.Errorf("error listing tags of Sagemaker endpoint configuration %s: %s", d.Id(), err)
+	}
+	if err := d.Set("tags", tagsToMapSagemaker(tagsOutput.Tags)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func resourceAwsSagemakerEndpointConfigurationUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	d.Partial(true)
+
+	if err := setSagemakerTags(conn, d); err != nil {
+		return err
+	} else {
+		d.SetPartial("tags")
+	}
+
+	d.Partial(false)
+
+	return resourceAwsSagemakerEndpointConfigurationRead(d, meta)
+}
+
+func resourceAwsSagemakerEndpointConfigurationDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	deleteOpts := &sagemaker.DeleteEndpointConfigInput{
+		EndpointConfigName: aws.String(d.Id()),
+	}
+	log.Printf("[INFO] Deleting Sagemaker endpoint configuration: %s", d.Id())
+
+	return resource.Retry(5*time.Minute, func() *resource.RetryError {
+		_, err := conn.DeleteEndpointConfig(deleteOpts)
+		if err == nil {
+			return nil
+		}
+
+		sagemakerErr, ok := err.(awserr.Error)
+		if !ok {
+			return resource.NonRetryableError(err)
+		}
+
+		if sagemakerErr.Code() == "ResourceNotFound" {
+			return resource.RetryableError(err)
+		}
+
+		return resource.NonRetryableError(fmt.Errorf("Error deleting Sagemaker endpoint configuration: %s", err))
+	})
+}
+
+func expandProductionVariants(configured []interface{}) ([]*sagemaker.ProductionVariant, error) {
```

Comment:
Two things:
* This function never returns an `error` so that should be removed from the function signature
* Nit: Since this codebase uses a very flat Go package structure, we should include the AWS service name in the expand and flatten function names

```suggestion
func expandSagemakerProductionVariants(configured []interface{}) []*sagemaker.ProductionVariant {
```